### PR TITLE
dependencies plugin

### PR DESCRIPTION
**Pull request by iignatevich**

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			if toPath {
+				res = convertToPath(res)
+			}
+
+			cli.Println("%s", res)
+		}
+	}
+
+	return nil
+}
```
Comment (davidferlay) [inline plugin.go:129]:
```
plasmactl build -p github.com/launchrctl/launchr@latest,github.com/launchrctl/compose@latest,github.com/skilld-labs/plasmactl-bump@latest,github.com/skilld-labs/plasmactl-package@latest,github.com/skilld-labs/plasmactl-publish@latest,github.com/skilld-labs/plasmactl-meta@latest,github.com/skilld-labs/plasmactl-update@latest,github.com/skilld-labs/plasmactl-dependencies@dependencies_plugin -n plasmactl -o plasmactl --build-version v0.17.3
```
Returns error:
```
go: added github.com/skilld-labs/plasmactl-publish v0.0.0-20240304155229-a91ed8fa1bf1
# github.com/skilld-labs/plasmactl-dependencies
../../../go/pkg/mod/github.com/skilld-labs/plasmactl-dependencies@v0.0.0-20240402131248-f84d1c6ebaa0/plugin.go:70:21: undefined: plasmactlbump.BuildResourceFromPath
../../../go/pkg/mod/github.com/skilld-labs/plasmactl-dependencies@v0.0.0-20240402131248-f84d1c6ebaa0/plugin.go:113:25: inv.GetDependenciesMap undefined (type *plasmactlbump.Inventory has no field or method GetDependenciesMap)
main.go:2: running "go": exit status 1
Error: exit status 1
```

**Comment by davidferlay** [comment]
- `--mrn` option works as expected :ok_hand: 
```
➜ ./plasmactl dependencies interaction/applications/roles/repositories 
- Parent dependencies:
interaction/services/roles/repositories_nexus

➜ ./plasmactl dependencies interaction/applications/roles/repositories --mrn 
- Parent dependencies:
interaction__services__repositories_nexus

```

File: plugin.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	inv, err := plasmactlbump.NewInventory("empty", ".", "")
+	if err != nil {
+		return err
+	}
+	requiredMap := inv.GetRequiredMap()
+	children, ok := requiredMap[searchMrn]
+
+	if ok {
+		cli.Println("- Child dependencies:")
```
Comment (davidferlay) [inline plugin.go:117]: - Typo: 
   - `Child dependencies:` > `Parent dependencies:`
   - `Parent dependencies:` > `Child dependencies:`

```
➜ ./plasmactl dependencies interaction/services/roles/repositories_nexus
- Child dependencies:
interaction/applications/roles/repositories
- Parent dependencies:
interaction/softwares/roles/nexus
```

File: plugin.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+			if toPath {
+				res = convertToPath(res)
+			}
+
+			cli.Println("%s", res)
+		}
+	}
+
+	dependenciesMap := inv.GetDependenciesMap()
+	parents, ok := dependenciesMap[searchMrn]
```
Comment (davidferlay) [inline plugin.go]: Output should recursively list dependencies of dependencies; as well as required of required

- Observed:
```
➜ ./plasmactl dependencies interaction/applications/roles/repositories 
- Child dependencies:
interaction/services/roles/repositories_nexus
```
- Expected:

```
➜ ./plasmactl dependencies interaction/applications/roles/repositories 
- Child dependencies:
interaction/services/roles/repositories_nexus
interaction/softwares/roles/nexus
```

**Comment by davidferlay** [comment]
- Would be great for command to also work in build dir, as some dependencies will only be resolved there
```
➜ ./plasmactl dependencies .compose/build/foundation/applications/roles/pki/                                                       
Error: not valid resource ".compose/build/foundation/applications/roles/pki/"                                               
```

File: plugin.go (hunk 3 above)
Comment (davidferlay) [reply plugin.go]: for instance with `➜ ./plasmactl dependencies interaction/applications/roles/mail`
we could get 
```

- Child dependencies:
cognition/softwares/roles/druid_exporter
cognition/softwares/roles/nats_kafka
foundation/services/roles/cluster_cni
foundation/softwares/roles/cni
foundation/softwares/roles/couchbase_exporter
foundation/softwares/roles/flannel_cni
foundation/softwares/roles/ingress_ldap_auth
foundation/softwares/roles/kube_certs_renew
foundation/softwares/roles/kubeadm
foundation/softwares/roles/kubectl
foundation/softwares/roles/kubelet
foundation/softwares/roles/openldap
foundation/softwares/roles/openldap_exporter
foundation/softwares/roles/prometheus_alert_machine
foundation/softwares/roles/prometheus_kafka_adapter
interaction/applications/roles/academy
interaction/applications/roles/landing
interaction/services/roles/account_alpine
interaction/services/roles/kb_alpine
interaction/services/roles/mail_dkimpy
interaction/services/roles/mail_dovecot
interaction/services/roles/mail_dovecot_exporter
interaction/services/roles/mail_postfix
interaction/services/roles/mail_postfix_exporter
interaction/services/roles/mail_rainloop
interaction/softwares/roles/dkimpy
interaction/softwares/roles/dovecot
interaction/softwares/roles/dovecot_exporter
interaction/softwares/roles/dovecot_ldap
interaction/softwares/roles/dovecot_ldap_sieve
interaction/softwares/roles/errors_server
interaction/softwares/roles/gitlab_api_extension
interaction/softwares/roles/gitlab_logger
interaction/softwares/roles/go_mmproxy
interaction/softwares/roles/grafana
interaction/softwares/roles/grafana_alert_machine
interaction/softwares/roles/openssh
interaction/softwares/roles/postfix
interaction/softwares/roles/postfix_exporter
interaction/softwares/roles/postfix_ldap
interaction/softwares/roles/rainloop
interaction/softwares/roles/skilld_account_machine
platform/softwares/roles/alpine
platform/softwares/roles/c
platform/softwares/roles/fetcher
platform/softwares/roles/golang
platform/softwares/roles/java
platform/softwares/roles/maven
platform/softwares/roles/nginx
platform/softwares/roles/npm
platform/softwares/roles/php
```

Comment (davidferlay) [reply plugin.go]: and with `➜ ./plasmactl dependencies interaction/applications/roles/mail` 
we could get 
```
- Parent dependencies:
interaction/applications/roles/mail
- Child dependencies:
foundation/services/roles/cluster_cni
foundation/softwares/roles/cni
foundation/softwares/roles/flannel_cni
foundation/softwares/roles/ingress_ldap_auth
foundation/softwares/roles/kube_certs_renew
foundation/softwares/roles/kubeadm
foundation/softwares/roles/kubectl
foundation/softwares/roles/kubelet
foundation/softwares/roles/openldap
foundation/softwares/roles/prometheus_kafka_adapter
interaction/applications/roles/academy
interaction/applications/roles/landing
interaction/services/roles/account_alpine
interaction/services/roles/kb_alpine
interaction/softwares/roles/dkimpy
interaction/softwares/roles/dovecot
interaction/softwares/roles/go_mmproxy
interaction/softwares/roles/openssh
interaction/softwares/roles/postfix
interaction/softwares/roles/postfix_ldap
platform/softwares/roles/alpine
platform/softwares/roles/c
platform/softwares/roles/fetcher
platform/softwares/roles/java
platform/softwares/roles/maven
platform/softwares/roles/nginx
platform/softwares/roles/npm
platform/softwares/roles/php
```

Comment (davidferlay) [reply plugin.go]: I'm thinking maybe we should add an option like `--depth=n` to limit recursion, if needed by user

And  by default list all

Comment (iignatevich) [reply plugin.go]: done

File: plugin.go (hunk 2 above)
Comment (iignatevich) [reply plugin.go:117]: swapped

**Comment by davidferlay** [comment]
`--depth` option behaves a bit weirdly: 
- Steps:
```
➜ ./plasmactl dependencies --depth=1 interaction/applications/roles/mail
```
- Observed:
```
- Child dependencies:
interaction/services/roles/mail_dkimpy
interaction/services/roles/mail_dovecot
interaction/services/roles/mail_dovecot_exporter
interaction/services/roles/mail_postfix
interaction/services/roles/mail_postfix_exporter
interaction/services/roles/mail_rainloop
interaction/softwares/roles/dkimpy
interaction/softwares/roles/dovecot_exporter
interaction/softwares/roles/dovecot_ldap_sieve
interaction/softwares/roles/postfix_exporter
interaction/softwares/roles/postfix_ldap
interaction/softwares/roles/rainloop
```
- Expected:
```
- Child dependencies:
interaction/services/roles/mail_dkimpy
interaction/services/roles/mail_dovecot
interaction/services/roles/mail_dovecot_exporter
interaction/services/roles/mail_postfix
interaction/services/roles/mail_postfix_exporter
interaction/services/roles/mail_rainloop
```

**Pull request merged**